### PR TITLE
Homework string mau

### PR DESCRIPTION
**Pull request by zerokill**
Done with homework. I am not quite sure about the amount of c code in there. eg determining the length of the string using pointers.

Or was I supposed to use c++ arrays instead of c arrays?

Regards,

Maurice

### CODE REVIEW

File: course3/homework-string/my_string.h [hunk 1]
```diff
@@ -2,5 +2,73 @@
...
+
+    bool empty() {
+        return length == 0 ? true : false;
+    }
+
+    int size() const {
+        return length;
+    }
+
+    int size() {
```
Comment (maciekgajewski) [inline course3/homework-string/my_string.h]: what's the use of non-const size()?

File: course3/homework-string/my_string.h [hunk 2]
```diff
@@ -2,5 +2,73 @@
...
+
+public:
+    MyString() {
+        length = 0;
+    }
+
+    MyString(const char *data) {
+        length = strlen(data);
+
+        buff = new char[length];
```
Comment (maciekgajewski) [inline course3/homework-string/my_string.h]: Where do you free the memory?

**Comment by maciekgajewski** [comment]
There is way too much C indeed. If you re-read my tips, you'll notice that I recommend the use of std::strlen and std::strcmp functions to measure and compare the string.
Also, you allocate memory, but you never delete it. Run your program trough valgrind, and you'll see!

File: course3/homework-string/my_string.h [hunk 3]
```diff
@@ -1,6 +1,53 @@
...
+    int size() const {
+        return length;
+    }
+
+    bool operator!=(const MyString& rhs) {
+        return !operator==(rhs);
+    }
+
+    bool operator==(const MyString& rhs) {
+        return std::strcmp(buff.get(), rhs.buff.get()) == 0;
```
Comment (zerokill) [inline course3/homework-string/my_string.h]: Question about this: How am i able to access a private of rhs? Does the compiler recognise that this function is called in the class method of this private?

**Pull request closed without merging**